apis/project.cattle.io/v3/schema/registry_credential.go

```diff
@@ -18,17 +19,33 @@ func (e RegistryCredentialMapper) ToInternal(data map[string]interface{}) error
 		return nil
 	}
 
-	auth := convert.ToString(data["auth"])
-	username := convert.ToString(data["username"])
-	password := convert.ToString(data["password"])
-
-	if auth == "" && username != "" && password != "" {
-		data["auth"] = base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+	if data["kind"] != "dockerCredential" {
+		return nil
 	}
 
+	addAuthInfo(data)
+
 	return nil
 }
 
+func addAuthInfo(data map[string]interface{}) error {
+
+	registryMap := convert.ToMapInterface(data["registries"])
+	for _, regCred := range registryMap {
+		regCredMap := convert.ToMapInterface(regCred)
+
+		username := convert.ToString(regCredMap["username"])
+		password := convert.ToString(regCredMap["password"])
+
+		if username == "" || password == "" {
```

Comment:
Nit: we can check the username right after line 37, and return if empty. this way we save on the conversion call for the password